source/common/http/http3/conn_pool.cc

```diff
@@ -133,14 +145,24 @@ allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_
         Network::Connection& connection = *data.connection_;
         auto client = std::make_unique<ActiveClient>(*pool, data);
         if (connection.state() == Network::Connection::State::Closed) {
+          ASSERT(dynamic_cast<CodecClientProd*>(client->codec_client_.get()) != nullptr);
```

Comment:
nit: I guess I'd add a note here about removing this conditional when the runtime flag is removed, but I don't think it gates approval.

Reply:
done